File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Support.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Support extends BaseResource {

    public Support(HttpClient http) {
        super(http);
    }

    /** List tickets */
    public Map<String, Object> tickets() throws Exception {
        return http.get("/api/tickets");
    }

    /** Create ticket */
    public Map<String, Object> createTicket(Map<String, Object> data) throws Exception {
        return http.post("/api/tickets", data);
    }

    /** Get ticket */
    public Map<String, Object> ticket(String number) throws Exception {
        return http.get(String.format("/api/tickets/%number", number));
    }

    /** Reply to ticket */
    public Map<String, Object> replyTicket(String number, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/tickets/%number", number), data);
    }

    /** Get attachment */
    public Map<String, Object> ticketAttachment(String file) throws Exception {
        return http.get(String.format("/api/ticket/attachment/%file", file));
    }

    /** Open ticket */
    public Map<String, Object> openTicket(String number) throws Exception {
        return http.put(String.format("/api/tickets/%number/open", number), null);
    }

    /** Close ticket */
    public Map<String, Object> closeTicket(String number) throws Exception {
        return http.put(String.format("/api/tickets/%number/close", number), null);
    }

    /** Get departments */
    public Map<String, Object> departments() throws Exception {
        return http.get("/api/ticket/departments");
    }

    /** List news */
    public Map<String, Object> news() throws Exception {
        return http.get("/api/news");
    }

    /** Get news item */
    public Map<String, Object> newsItem(int newsId) throws Exception {
        return http.get(String.format("/api/news/%newsId", newsId));
    }

    /** List knowledgebase */
    public Map<String, Object> knowledgebase() throws Exception {
        return http.get("/api/knowledgebase");
    }

    /** Get KB category */
    public Map<String, Object> knowledgebaseCategory(int categoryId) throws Exception {
        return http.get(String.format("/api/knowledgebase/%categoryId", categoryId));
    }

    /** Get KB article */
    public Map<String, Object> knowledgebaseArticle(int articleId) throws Exception {
        return http.get(String.format("/api/knowledgebase/article/%articleId", articleId));
    }

}

File: sdk/nodejs/src/resources/support.js
"use strict";

const { BaseResource } = require("./base");

class SupportResource extends BaseResource {
  async listTickets(params = {}) {
    return this._http.get("/api/tickets", params);
  }

  async createTicket(data) {
    return this._http.post("/api/tickets", data);
  }

  async getTicket(ticketNumber) {
    return this._http.get(`/api/tickets/${ticketNumber}`);
  }

  async reply(ticketNumber, message) {
    return this._http.post(`/api/tickets/${ticketNumber}`, { message });
  }

  async getAttachment(file) {
    return this._http.get(`/api/ticket/attachment/${file}`);
  }

  async reopen(ticketNumber) {
    return this._http.put(`/api/tickets/${ticketNumber}/open`);
  }

  async close(ticketNumber) {
    return this._http.put(`/api/tickets/${ticketNumber}/close`);
  }

  async listDepartments() {
    return this._http.get("/api/ticket/departments");
  }

  async listNews() {
    return this._http.get("/api/news");
  }

  async getNews(newsId) {
    return this._http.get(`/api/news/${newsId}`);
  }

  async listKnowledgebaseCategories() {
    return this._http.get("/api/knowledgebase");
  }

  async getKnowledgebaseCategory(categoryId) {
    return this._http.get(`/api/knowledgebase/${categoryId}`);
  }

  async getKnowledgebaseArticle(articleId) {
    return this._http.get(`/api/knowledgebase/article/${articleId}`);
  }
}

module.exports = { SupportResource };
